src/libfetchers/git.cc

```diff
@@ -191,25 +191,151 @@ struct GitInputScheme : InputScheme
         return "git";
     }
 
-    StringSet allowedAttrs() const override
+    std::string schemeDescription() const override
+    {
+        return stripIndentation(R"(
+          Fetch a Git tree and copy it to the Nix store.
+          This is similar to [`builtins.fetchGit`](@docroot@/language/builtins.md#builtins-fetchGit).
+        )");
+    }
+
+    StringMap allowedAttrs() const override
     {
         return {
-            "url",
-            "ref",
-            "rev",
-            "shallow",
-            "submodules",
-            "lastModified",
-            "revCount",
-            "narHash",
-            "allRefs",
-            "name",
-            "dirtyRev",
-            "dirtyShortRev",
-            "verifyCommit",
-            "keytype",
-            "publicKey",
-            "publicKeys",
+            {
+                "url",
+                R"(
+                  (String, required)
+
+                  The URL formats supported are the same as for Git itself.
+
```

Comment:
There should be a struct with `type`. `optional`, and `doc`, fields, similar for that for primops.